src/pip/_internal/operations/prepare.py

```diff
@@ -568,7 +581,10 @@ def prepare_linked_requirements_more(
         )
 
     def _prepare_linked_requirement(
-        self, req: InstallRequirement, parallel_builds: bool
+        self,
+        req: InstallRequirement,
+        parallel_builds: bool,
+        source_dir_exists_already: bool = False,
```

Comment:
Is the `False` default necessary? Having a default value for internal functions feels like an anti-pattern.

Reply:
Totally agree, will remove this.